src/codegen/mod.rs

```diff
@@ -2585,9 +2585,25 @@ impl CodeGenerator for Enum {
         }
 
         if !variation.is_const() {
-            attrs.push(attributes::derives(
-                &["Debug", "Copy", "Clone", "PartialEq", "Eq", "Hash"],
-            ));
+            let mut derives = vec!["Debug", "Copy", "Clone", "Hash"];
```

Comment:
Looks great, thanks! Let's keep `Eq` and `PartialEq` here for backwards compat instead of checking, since its what we were doing.

Also, a test must be added. See all the tests that use `--with-derive-ord` in `tests/headers`.

Reply:
@emilio many thanks!!

When moving `PartialEq` and `Eq` out of the default list, my intention is to handle `--no-partialeq`. For example,

```
enum A {
	a0 = 0,
	a1 = 1
};

enum B {
	b0 = 1,
	b1 = 2
};
```

then with the command:

```
bindgen test.h --with-derive-partialeq --rustified-enum "*" --no-partialeq "B"
```

we will get:

```
#[repr(i32)]
#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
pub enum A {
    a0 = 0,
    a1 = 1,
}
#[repr(i32)]
#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
pub enum B {
    b0 = 1,
    b1 = 2,
}
```

In fact, there is currently no way to apply `--no-partialeq` on `Enum`. So that is the reason why I moved `PartialEq` and `Eq` out, and for the same command I get

```
#[repr(i32)]
#[derive(Debug, Copy, Clone, Hash, PartialEq)]
pub enum A {
    a0 = 0,
    a1 = 1,
}
#[repr(i32)]
#[derive(Debug, Copy, Clone, Hash)]
pub enum B {
    b0 = 1,
    b1 = 2,
}
```

What do you think? If you prefer the previous behavior for backward compatibility, I will update the PR.

NB. I will add tests too.

Reply:
I... Don't really have that much of a strong opinion, though I just suspect `PartialEq` is pretty universally useful in enums, but is tricky on structs (since pointer comparison is not deep, etc).

Given there's no way to "enable partialeq on enums, disable for structs", I think I'd prefer the previous behavior for now.

Perhaps it's worth a different discussion, or an API to allow enabling derives separately for enums / structs (which is actually trivial to implement).